packit_service/worker/handlers/abstract.py

```diff
@@ -161,12 +174,47 @@ class JobHandler(Handler):
     triggers: List[TheJobTriggerType]
 
     def __init__(
-        self, config: ServiceConfig, job_config: Optional[JobConfig], event: Event
+        self,
+        package_config: PackageConfig,
+        job_config: Optional[JobConfig],
+        event: dict,
     ):
-        super().__init__(config)
+        self.package_config: PackageConfig = package_config
         self.job_config: Optional[JobConfig] = job_config
+        self.event_type: str = event.get("event_type")
+        self.trigger: TheJobTriggerType = TheJobTriggerType(event.get("trigger"))
+        self.user_login: str = event.get("user_login")
+        self.trigger_id: int = event.get("trigger_id")
+        self.project_url: str = event.get("project_url")
```

Comment:
>  @lbarcziova you created a new class for those metadata, didn't you?
Cannot be that enhanced to server also there?
I would like to avoid dicts for the same reason as I wrote in build-helpers: typing, debugging,...

@lachmanfrantisek I don't know if I understand how you mean it, in the JobHandler there are these attributes needed almost by each handler parsed from the dict

Reply:
I have the same problems with using dictionary representation as I have with it in helper classes:
- debugging is harder
- types cannot be (easily) checked
- code is less readable (a lot of `get` noise and strings can be mistyped)
- autocompletion does not work (and other IDE helpers)

Two possible solutions:
- Reuse and enhance the metadata class you already created.
- Create a new class that will handle those values.

Reply:
I got your point, I was just thinking that now each handler in init calls the super(JobHandler) init method so in JobHandler, the common attributes (event_type, trigger_id, trigger, user_login, project_url) are parsed (and optionally some specific attributes in own init method are parsed), therefore no simple way came to my mind to get rid of the dictionary totally.

Would something like this make sense? Having new class with the common attributes as I wrote above, having classmethod in which the parsing would be done as in the new TriggerMetadata class and then in the handlers it would look like this:
```python
class ProposeDownstreamHandler(JobHandler):
    type = JobType.propose_downstream
    triggers = [TheJobTriggerType.release]

    def __init__(
        self,
        package_config: PackageConfig,
        job_config: Optional[JobConfig],
        handler_data: HandlerData,
        tag_name: str,
    ):
        super().__init__(
            package_config=package_config, job_config=job_config, data=handler_data,
        )
        self.tag_name = tag_name
        
    @classmethod
    def from_event_dict(
        cls,
        package_config: PackageConfig,
        job_config: Optional[JobConfig],
        event: dict,
    ):
        tag_name = event.get("tag_name")
        return ProposeDownstreamHandler(
            package_config=package_config,
            job_config=job_config,
            handler_info=HandlerData.from_event_dict(event)
            tag_name=tag_name
        )
```

Reply:
I see that it's not the same case.

- If the attributes are mostly the same, I would add everything to the `HandlerData` and make the specific ones optional. An instance of `HandlerData` can then be instantiated before handler and send as an instance (not a dictionary) to the `__init__` of the handler. (We will need to fight with the 'Optional` there.)
- If the attributes differ a lot we can have multiple types of `HandlerData` (a lot of work), or leave it as it is now. (I am not so against now..;)
- Or, we can implement that as `kwargs` that will be forwarded to the `__init__` of handlers. (You will send  `**event` when instantiating the handler.)

Reply:
So, I created a class for the common attributes and since more of the TriggerData attributes were required there, I removed the TriggerData class. Nevertheless, for some handlers (e.g.Koji/CoprReportHandler), there is also event dict passed, as the attributes there differ from the most of the handlers, but in the following PR, where the tasks will be implemented, these specific attributes can be parsed before handler instantiation as the handlers would be called separately by tasks.